index.js

```diff
@@ -0,0 +1,65 @@
+let button1 = document.querySelector('#fourx');
+let button2 = document.querySelector('#thirtytwox');
+let button3 = document.querySelector('#img');
+let canvas = document.querySelector('#canvas');
+
+function draw1(arr) {
+    if (canvas.getContext) {
+        var ctx = canvas.getContext('2d');
+
+        arr.forEach((row, i) => {
+            row.forEach((column, j) => {
+                ctx.fillStyle = "#" + column;
+                ctx.fillRect(j * (512 / arr.length), i * (512 / arr.length), (j + 1) * (512 / arr.length), (i + 1) * (512 / arr.length));
+            })
+        })
+    }
+}
+
+function draw2(arr) {
+    if (canvas.getContext) {
+        var ctx = canvas.getContext('2d');
+
+        arr.forEach((row, i) => {
+            row.forEach((column, j) => {
+                ctx.fillStyle = "rgba(" + column + ")";
+                ctx.fillRect(j * (512 / arr.length), i * (512 / arr.length), (j + 1) * (512 / arr.length), (i + 1) * (512 / arr.length));
+            })
+        })
+    }
+}
```

Comment:
@VikinGSpb Functions `draw1` and `draw2` looks the same. You can reuse it instead of duplicating.